### PR TITLE
In XSTR update and DRY the scenario data

### LINKED ISSUE #367: DRY datasets: `pstr_scenarios` and `istr_senarios`
Both datasets are identical so they could be called simply 'scenarios'. This
is similar to what we do with 'companies', 'inputs', and 'products'.

``` r
devtools::load_all()
#> ℹ Loading tiltIndicator
packageVersion("tiltIndicator")
#> [1] '0.0.0.9064'

identical(istr_scenarios, pstr_scenarios)
#> [1] TRUE
```

<sup>Created on 2023-05-30 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

### LINKED ISSUE #387: Change column names from `weo_product`, `weo_flow` to `weo_sector`, `weo_subsector` in the `str_weo_targets` scenario dataset

### LINKED ISSUE #388: Update the scenario datasets for ISTR and PSTR
Scenario datasets `str_ipr_targets` and `str_weo_targets` must be updated from this source: https://drive.google.com/drive/folders/15AsbEhET2WvU0z2dFhyKfvZF9jQHPuvP

### PR DESCRIPTION
**Pull request by kalashsinghal**
closes #367 
closes #388 
closes #387 

Dear @maurolepore 
I have updated the scenario datasets for ISTR and PSTR after getting the new dataset from Tilman through slack. Also, I have made a new R script titled "xstr_scenarios.R" to replace the `pstr_scenarios` and `istr_scenarios` datasets as they both are similar. 

Thanks for your review!

----

TODO

- [x] [Link related issue/PR]([url](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)). 
- [x] Describe the goal of the PR. Avoid details that are clear in the diff.
- [x] Mark the PR as draft.
- [x] Review your own PR in "Files changed".
- [x] Ensure the PR branch is updated.
- [x] [Include a unit test](https://code-review.tidyverse.org/reviewer/aspects.html#sec-tests).
- [x] Ensure the checks pass.
- [x] Change the status from draft to ready.
- [x] Polish the PR title and description.
- [x] Assign a reviewer.

EXCEPTIONS

- [ ] Slide here any item that you intentionally choose to not do.

### CODE REVIEW

File: R/xstr_prepare_scenario.R [hunk 1]
```diff
@@ -19,15 +19,6 @@
...
 #' xstr_prepare_scenario(raw_scenarios)
 xstr_prepare_scenario <- function(scenarios) {
-  # Hack #308
-  if (hasName(scenarios, "weo")) {
-    scenarios$weo <- rename(
-      scenarios$weo,
-      weo_sector = "weo_product",
-      weo_subsector = "weo_flow"
-    )
-  }
```
Comment (kalashsinghal) [inline R/xstr_prepare_scenario.R:29]: @maurolepore I have updated the `weo` column names in the original datasets. Hence, now there is no need to change the column names inside the code.

File: tests/testthat/test-istr.R [hunk 2]
```diff
@@ -180,7 +180,7 @@ test_that("grouped_by includes the type of scenario", {
 test_that("with type ipr, for each company and grouped_by value sums 1 (#216)", {
   .type <- "ipr"
   companies <- istr_companies |> slice(1)
-  scenarios <- istr_scenarios |> filter(type == .type)
+  scenarios <- xstr_scenarios |> filter(type == .type)
   inputs <- istr_inputs |> filter(type == .type)
 
   out <- istr(companies, scenarios, inputs)
```
Comment (kalashsinghal) [inline tests/testthat/test-istr.R:186]: @maurolepore this test is failing without any good reason. The output of the second line should be`TRUE`. Please correct this and similar test! Thanks!
```
> c(na.omit(sum$value_sum))[1]
[1] 1
> c(na.omit(sum$value_sum))[1] == 1
[1] FALSE
```

**Review by maurolepore — APPROVED** [review]
LGTM. Thanks!

I pushed a few commits and I expect the checks to now pass.

File: _pkgdown.yml [hunk 3]
```diff
@@ -4,17 +4,21 @@ template:
...
 
+  - title: XSTR functions and datasets
+    contents:
       - has_concept("ISTR functions")
-      - has_concept("ISTR datasets")
-
       - has_concept("PSTR functions")
+
+      - has_concept("XSTR datasets")
+      - has_concept("ISTR datasets")
```
Comment (maurolepore) [inline _pkgdown.yml:21]: Sorry I took the change to rearrange the website reference a bit. This doesn't quite belong to this PR but doesn't seem to interfere as it's such a different type of change.

File: tests/testthat/test-istr.R (hunk 2 above)
Comment (kalashsinghal) [reply tests/testthat/test-istr.R:186]: @maurolepore Thanks for resolving this issue! I learned something new

File: data-raw/xstr_scenarios.R [hunk 4]
```diff
@@ -0,0 +1,13 @@
+# Source: Tilman via slack
```
[the remark this answers is not in the document]
Comment (kalashsinghal) [reply data-raw/xstr_scenarios.R]: Added through https://github.com/2DegreesInvesting/tiltIndicator/pull/413/commits/8d226b1729def9a3ab2a6ae92fbb20872f063c9f

**Pull request merged**